pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1001,3 +1013,94 @@ func (b *SlirpPodInterface) setCachedVIF(pid, name string) error {
 func (s *SlirpPodInterface) setCachedInterface(pid, name string) error {
 	return nil
 }
+
+type MacvtapPodInterface struct {
+	vmi              *v1.VirtualMachineInstance
+	vif              *VIF
+	iface            *v1.Interface
+	virtIface        *api.Interface
+	domain           *api.Domain
+	podInterfaceName string
+}
+
+func (m *MacvtapPodInterface) discoverPodNetworkInterface() error {
+	if len(m.vif.MAC) == 0 {
+		// Get interface MAC address
+		mac, err := Handler.GetMacDetails(m.podInterfaceName)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to get MAC for %s", m.podInterfaceName)
+			return err
+		}
+		m.vif.MAC = mac
+	}
+
+	return nil
+}
+
+func (m *MacvtapPodInterface) preparePodNetworkInterfaces() error {
+	m.virtIface.MAC = &api.MAC{MAC: m.vif.MAC.String()}
+	m.virtIface.Target = &api.InterfaceTarget{
```

Comment:
Why is setting the target on the domain done here and not in the `converter`? AFAIU, looking on other examples, the `converter` is setting on the domain all the information that can be taken from the spec. The `podInterface` is setting the extra information that needs interaction with the nics (e.g mtu, mac). Here only the macvtap interface name is needed, the converter has this [information](https://github.com/kubevirt/kubevirt/blob/aa144fedd101202997427d17ac8d3624cf08f54d/pkg/virt-launcher/virtwrap/api/converter.go#L1239).

Reply:
I was talking about the `Target` attribute, not the mac address. Now I see you put the `target` tap device for maquerade binding here as well. We can keep it here for consistency. Up to you.

Reply:
I'll leave it then.